packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,262 @@
+| name          | `string`                    | No       | `undefined` | The name of the checkbox group, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name) |
+| value         | `string[]`                  | No       | `[]`        | The value of the checkbox, if present will act as controlled component                                                             |
+| onChange      | `(value: string[]) => void` | No       | `undefined` | The function to be called when any checkbox's state changes                                                                        |
+| isDisabled    | `boolean`                   | No       | `false`     | Control whether the checkbox group is disabled or not.                                                                             |
+| isReadOnly    | `boolean`                   | No       | `false`     | Control whether the checkbox group is readonly or not.                                                                             |
+| isInvalid     | `boolean`                   | No       | `false`     | Control whether the checkbox group is invalid or not.                                                                              |
+| isRequired    | `boolean`                   | No       | `false`     | Control whether the checkbox group is required or not.                                                                             |
+
+
+
+### Examples:
+
+#### Basic
+
+```tsx
+<CheckboxGroup
+  label="Developers"
+  helperText="Pick blade developers"
+  defaultValue={['anurag', 'kamlesh']}
+>
+  <Checkbox value="anurag">Anurag</Checkbox>
+  <Checkbox value="kamlesh">Kamlesh</Checkbox>
+  <Checkbox value="chaitanya">Chaitanya</Checkbox>
+</CheckboxGroup>
+```
+
+#### Controlled, Uncontrolled
+
+```tsx
+const Controlled = () => {
+  const [selected, setSelected] = React.useState(['anurag', 'kamlesh'])
+
+  return (
+    <CheckboxGroup
+      label="Developers (controlled)"
+      value={selected}
+      onChange={setSelected}
+    >
+    <Checkbox value="anurag">Anurag</Checkbox>
+    <Checkbox value="kamlesh">Kamlesh</Checkbox>
+    <Checkbox value="chaitanya">Chaitanya</Checkbox>
+  </CheckboxGroup>
+  )
+}
+
+const Uncontrolled = () => {
+  return (
+    <CheckboxGroup
+      label="Developers (uncontrolled)"
+      defaultValue={['anurag', 'chaitanya']}
+    >
+      <Checkbox value="anurag">Anurag</Checkbox>
+      <Checkbox value="kamlesh">Kamlesh</Checkbox>
+      <Checkbox value="chaitanya">Chaitanya</Checkbox>
+    </CheckboxGroup>
+  )
+}
+```
+
+
+### Note on CheckboxGroup
+
+For properties `isDisabled`, `isReadOnly`, `isInvalid`, `isRequired` we want the CheckboxGroup to pass down these props to all the Checkboxes inside of it, but there is a gotcha. 
```

Comment:
Should we pass these props from the `CheckboxGroup` or at individual 'Checkbox'? I don't remember the last time when I saw all the checkboxes in a group being disabled, readonly or invalid at once. Generally individual checkboxes can have those states.

Reply:
Both will be supported you can pass them to Group or individually too. Since <Checkbox /> itself can also have all those common properties.